### PR TITLE
Alcail/participants button story

### PR DESCRIPTION
**Pull request by alcail**
# What
added ParticipantsButton story and docs
image
image
image
image

added ParticipantsButton to ControlBar story and docs

# Why
task https://skype.visualstudio.com/SPOOL/_workitems/edit/2460642

# How Tested
ran storybook locally

# Process & policy checklist

- [x] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [x] This change causes current functionality to break.
added ButtonCustomStylesProps as new type for use in ParticipantsButton

### CODE REVIEW

File: packages/storybook/stories/ControlBar/Buttons/Participants/Participants.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+// Licensed under the MIT license.
+
+import { ParticipantsButton, ParticipantListProps } from '@azure/communication-react';
+import { Canvas, Description, Heading, Props, Source, Title } from '@storybook/addon-docs/blocks';
+import { boolean, text } from '@storybook/addon-knobs';
+import { Meta } from '@storybook/react/types-6-0';
+import React from 'react';
+
+import { COMPONENT_FOLDER_PREFIX } from '../../../constants';
+// import { OptionsButtonCustomExample } from './snippets/Custom.snippet';
```
Comment (mgamis-msft) [inline packages/storybook/stories/ControlBar/Buttons/Participants/Participants.stories.tsx:11]: Did you accidentally leave in this commented import and the one on line 20?

File: packages/react-components/src/components/ParticipantsButton.tsx [hunk 2]
```diff
@@ -67,9 +67,9 @@ export interface ParticipantsButtonProps extends IButtonProps {
 
 /**
  * `ParticipantsButton` allows you to easily create a component rendering a participants button. It can be used in your ControlBar component for example.
- * This button, by default, contains dropdown menu items defined through its property `menuProps` (user presence, number of remote participants with a list
- * as sub-menu and a copy-to-clipboard button of the call invite URL).
- * This `menuProps` property is of type [IContextualMenuProps](https://developer.microsoft.com/en-us/fluentui#/controls/web/contextualmenu#IContextualMenuProps).
+ * This button contains dropdown menu items defined through its property `menuProps`. By default, it can display user presence, number of remote participants with a list
+ * as sub-menu and an options to mute all participants, as well as a copy-to-clipboard button to copy the call invitation URL.
```
Comment (mgamis-msft) [inline packages/react-components/src/components/ParticipantsButton.tsx:71]: Should this be '...a list as sub-menu and options to ...'?

**Review by mgamis-msft — APPROVED** [review]

File: packages/storybook/stories/ControlBar/Buttons/Participants/Participants.stories.tsx (hunk 1 above)
Comment (alcail) [reply packages/storybook/stories/ControlBar/Buttons/Participants/Participants.stories.tsx:11]: oopsi... forgot to remove those comments :)

File: packages/storybook/stories/ControlBar/ControlBar.stories.tsx [hunk 3]
```diff
@@ -163,6 +202,10 @@ const ControlBarStory: (
     }
   }
 
+  const onMuteAll = () => {
+    // your implementation to mute all participant
```
Comment (JamesBurnside) [inline packages/storybook/stories/ControlBar/ControlBar.stories.tsx:206]: nit: participants
```suggestion
    // your implementation to mute all participants
```

File: packages/storybook/stories/ControlBar/Buttons/Participants/snippets/WithCustomRender.snippet.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+export const ParticipantsButtonWithCustomRenderExample: () => JSX.Element = () => {
+  const customOnRenderIcon = (): JSX.Element => {
+    return <Icon key={'participantsCustomIconKey'} iconName={'Group'} style={{ color: 'orange', fontSize: '20px' }} />;
+  };
+
+  const customOnRenderText = (): JSX.Element => {
+    return (
+      <Label key={'participantsCustomLabelKey'} style={{ color: 'darkviolet', fontStyle: 'italic' }}>
+        who is an this call
```
Comment (JamesBurnside) [inline packages/storybook/stories/ControlBar/Buttons/Participants/snippets/WithCustomRender.snippet.tsx:86]: nit: an -> on
```suggestion
        Who is on this call
```

File: packages/storybook/stories/ControlBar/Buttons/Participants/snippets/WithCustomRender.snippet.tsx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+  participants: mockParticipants,
+  myUserId: 'user1',
+  excludeMe: true,
+  onRenderAvatar: (participant: CommunicationParticipant) => customOnRenderAvatar(participant)
+};
+
+const mockCallLink = 'URL to invite new participants to the current call (https://...)';
+
+const onMuteAll = () => {
+  // your implementation to mute all participant
```
Comment (JamesBurnside) [inline packages/storybook/stories/ControlBar/Buttons/Participants/snippets/WithCustomRender.snippet.tsx:75]: participant -> participants
```suggestion
  // your implementation to mute all participants
```

File: packages/storybook/stories/ControlBar/Buttons/Participants/Participants.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+import { ParticipantsButton } from '@azure/communication-react';
+`;
+
+const getDocs: () => JSX.Element = () => {
+  return (
+    <>
+      <Title>ParticipantsButton</Title>
+      <Description of={ParticipantsButton} />
+      <Description>
+        Note: When overriding a render, like using `onRenderIcon` or `onRenderText`, don not forget to add a unique key
```
Comment (JamesBurnside) [inline packages/storybook/stories/ControlBar/Buttons/Participants/Participants.stories.tsx:41]: nit: don -> do
```suggestion
        Note: When overriding a render, like using `onRenderIcon` or `onRenderText`, do not forget to add a unique key
```

File: packages/storybook/stories/ControlBar/ControlBar.stories.tsx (hunk 3 above)
Comment (JamesBurnside) [reply packages/storybook/stories/ControlBar/ControlBar.stories.tsx:206]: (one or two other places also)

File: packages/storybook/stories/ControlBar/Buttons/Participants/Participants.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+        sub-menu item of the number of participants, just below the list of participants, like shown in the example
+        below.
+      </Description>
+      <Canvas mdxSource={ParticipantsButtonWithMuteAllOptionExampleText}>
+        <ParticipantsButtonWithMuteAllOptionExample />
+      </Canvas>
+
+      <Heading>ParticipantsButton with cutom styles</Heading>
+      <Description>
+        You can change the styles of `ParticipantsButton` through its `styles` props as you would customized any Button
```
Comment (JamesBurnside) [inline packages/storybook/stories/ControlBar/Buttons/Participants/Participants.stories.tsx:95]: nit: customized - customize
```suggestion
You can change the styles of `ParticipantsButton` through its `styles` props as you would customize any Button
```

File: packages/storybook/stories/ControlBar/Buttons/Participants/Participants.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+        styles (root, label, etc..) with the addition of customizing the participant list container like shown in the
+        example below..
```
Comment (JamesBurnside) [inline packages/storybook/stories/ControlBar/Buttons/Participants/Participants.stories.tsx:97]: nit: one full stop:
```suggestion
        example below.
```

**Review by JamesBurnside — APPROVED** [review]
Loving all these snippets!

**Pull request merged**